mediapipe/tasks/ios/test/audio/audio_classifier/MPPAudioClassifierTests.mm

```diff
@@ -347,6 +398,121 @@ - (void)testClassifyFailsWithCallingWrongApiInAudioStreamMode {
   AssertEqualErrors(error, expectedError);
 }
 
+- (void)testClassifyWithAudioStreamModeAndOutOfOrderTimestampsFails {
+  MPPAudioClassifier *audioClassifier =
+      [self audioClassifierInStreamModeWithModelFileInfo:kYamnetModelFileInfo];
+  NSArray<MPPTimestampedAudioData *> *streamedAudioDataList =
+      [MPPAudioClassifierTests streamedAudioDataListforYamnet];
+
+  XCTestExpectation *expectation = [[XCTestExpectation alloc]
+      initWithDescription:@"classifyWithOutOfOrderTimestampsAndLiveStream"];
+  expectation.expectedFulfillmentCount = 1;
+
+  _outOfOrderTimestampTestDict = @{
+    kAudioStreamTestsDictClassifierKey : audioClassifier,
+    kAudioStreamTestsDictExpectationKey : expectation
+  };
+
+  // Can safely access indices 1 and 0 `streamedAudioDataList` count is already asserted.
+  XCTAssertTrue([audioClassifier
+      classifyAsyncAudioBlock:streamedAudioDataList[1].audioData
+      timestampInMilliseconds:streamedAudioDataList[1].timestampInMilliseconds
+                        error:nil]);
+
+  NSError *error;
+  XCTAssertFalse([audioClassifier
+      classifyAsyncAudioBlock:streamedAudioDataList[0].audioData
+      timestampInMilliseconds:streamedAudioDataList[0].timestampInMilliseconds
+                        error:&error]);
+
+  NSError *expectedError =
+      [NSError errorWithDomain:kExpectedErrorDomain
+                          code:MPPTasksErrorCodeInvalidArgumentError
+                      userInfo:@{
+                        NSLocalizedDescriptionKey :
+                            @"INVALID_ARGUMENT: Input timestamp must be monotonically increasing."
+                      }];
+  AssertEqualErrors(error, expectedError);
+
+  [audioClassifier closeWithError:nil];
+
+  NSTimeInterval timeout = 1.0f;
+  [self waitForExpectations:@[ expectation ] timeout:timeout];
+}
+
+- (void)testClassifyWithAudioStreamModeSucceeds {
+  [self testClassifyUsingYamnetAsyncAudioFileWithInfo:kSpeech16KHzMonoFileInfo
+                                                 info:&_16kHZAudioStreamSucceedsTestDict];
+  [self testClassifyUsingYamnetAsyncAudioFileWithInfo:kSpeech48KHzMonoFileInfo
+                                                 info:&_48kHZAudioStreamSucceedsTestDict];
+}
+
+// info is strong here since address of global variables will be passed to this function. By default
+// `NSDictionary **` will be `NSDictionary * __autoreleasing *.
+- (void)testClassifyUsingYamnetAsyncAudioFileWithInfo:(MPPFileInfo *)audioFileInfo
```

Comment:
Can we change the method name to make it a bit more clear that this is a test helper and not a test?

Reply:
Changed the name and added a new section for it.